File: tags/py-client/echo_1/echo.py
import sys

from echolib import *

HOST = 'agammemnon.csl.tjhsst.edu'
PORT = 45287

def main():
	c = Connection(HOST, PORT)
	var = c.connect()
	if var <= 0:
		if var is UNKNOWN_HOST:
			print 'Host \'%s\' not found.' % HOST
		elif var is CONNECTION_REFUSED:
			print 'Connection refused on port %d by host \'%s\'.' % (PORT, HOST)
		exit()

	print 'Connection successful.'

	while True:
		msg = raw_input()
		if msg == '.quit':
			break
		c.send(msg)
		print c.recv()

	print 'Cleaning up.'
	print 'Closing connection.'
	c.close()
	exit()

def exit():
	print 'Exiting.'
	sys.exit()

if __name__ == '__main__':
	main()

File: tags/py-client/echo_1/echo.java
import java.io.IOException;

import java.net.ConnectException;
import java.net.Socket;
import java.net.UnknownHostException;

public class echo{
	static final String HOST = "agammemnon.csl.tjhsst.edu";
	static final int PORT = 45287;

	public static void main(String[] args){
		Connection c = new Connection(HOST, PORT);
	}
}

class Connection{
	String host;
	int port;
	Socket socket;

	public Connection(String h, int p){
		host = h;
		port = p;
		try{
			socket = new Socket(h, p);
		}
		// TODO: Better error handling
		catch(UnknownHostException uhe){
			// XXX: This is a placeholder
			System.out.printf("Host '%s' not found. Exiting.\n", h);
			System.exit(0);
		}
		catch(ConnectException ce){
			System.out.printf("Connection refused on port %d. Exiting.\n", p);
			System.exit(0);
		}
		catch(IOException ioe){
			System.out.println("IOException caught. Exiting.");
			ioe.printStackTrace();
			System.exit(0);
		}

		System.out.println("Connection established!");
	}
}
